c_src/rocksdb_lib.cpp

```diff
@@ -444,9 +444,31 @@ rocksdb::Status PutTerms(db_obj_resource* rdb,
 			 rocksdb::Slice* value,
 			 std::vector<std::pair<Term, std::vector<Term>>> indices
 			) {
+    // prepare any new indices
     TermPrep tp = TermPrep(indices, key);
+    // delete any indicies that may be overwritten by this write
+    TermDelete td = TermDelete(tp.cids_, key);
```

Comment:
Can there be an optimisation like finding what exactly needs to be deleted? It can be at term preperation or maybe batch operations are already optimised by rocksdb?